gslab_scons/tests/test_size_warning.py

```diff
@@ -0,0 +1,312 @@
+        # Large over limit and total size over limit
+        with nostderrout():
+            sw.issue_size_warnings(look_in,
+                                   file_MB_limit = big_size - 0.1, 
+                                   total_MB_limit = total_size - 0.1)
+        self.assertEqual(mock_input.call_count, 2)
+        mock_input.reset_mock()    
+
+    @mock.patch('gslab_scons.size_warning.os.path.isfile')
+    @mock.patch('gslab_scons.size_warning.os.path.isdir')
+    @mock.patch('gslab_scons.size_warning.os.walk')
+    @mock.patch('gslab_scons.size_warning.subprocess.check_output')
+    def test_list_ignored_files(self, mock_check, mock_walk,
+                                mock_isdir, mock_isfile):
+        mock_check.side_effect  = check_ignored_side_effect('standard')
+        mock_walk.side_effect   = walk_ignored_side_effect
+        mock_isdir.side_effect  = isdir_ignored_side_effect
+        mock_isfile.side_effect = isfile_ignored_side_effect
+
+        look_in = ['raw', 'release']
+        ignored = sw.list_ignored_files(look_in)
+        expect_ignored = ['raw/large_file.txt',
+                          'release/.DS_Store', 
+                          'release/subdir/ignored.txt']
+        print ignored
+        self.assertEqual(len(ignored), len(expect_ignored))
+        for i in range(len(ignored)):
+            self.assertIn(ignored[i], expect_ignored)
+
+
+        look_in = ['raw']
+        ignored = sw.list_ignored_files(look_in)
+        expect_ignored = ['raw/large_file.txt']
+
+        self.assertEqual(len(ignored), len(expect_ignored))
+        self.assertEqual(ignored[0], expect_ignored[0])       
+
+        # Test that list_ignored_files returns an empty list when 
+        # git is not ignoring any files.
+        mock_check.side_effect  = check_ignored_side_effect(ignored = 'none')
+        ignored = sw.list_ignored_files(look_in)
+
+        self.assertIsInstance(ignored, list)
+        self.assertEqual(len(ignored), 0)
+
+    @mock.patch('gslab_scons.size_warning.os.path.getsize')
+    @mock.patch('gslab_scons.size_warning.os.walk')
+    @mock.patch('gslab_scons.size_warning.os.path.isdir')
+    def test_create_size_dictionary(self, mock_isdir, mock_walk, mock_getsize):
+        '''
+        Test that create_size_dictionary() correctly reports
+        files' sizes in bytes.
+        '''
+        # Assign side effects
+        mock_isdir.side_effect   = isdir_dict_side_effect        
+        mock_walk.side_effect    = walk_dict_side_effect
+        mock_getsize.side_effect = getsize_dict_side_effect 
+
+        # Test when one directory is provided
+        sizes = sw.create_size_dictionary(['test_files'])
+
+        self.assertEqual(len(sizes), 3)
+
+        # Check that test.txt and test.jpg are in the dictionary
+        root_path = [k for k in sizes.keys() if re.search('root_file.txt$', k)]
+        pdf_path  = [k for k in sizes.keys() if re.search('test.pdf$', k)]
+        self.assertTrue(bool(root_path))
```

Comment:
I might prefer a length check versus a boolean, as it seems odd to do boolean on a list. Applies throughout.